src/Views/ProgressView.vala

```diff
@@ -39,18 +44,35 @@ public class Sideload.ProgressView : AbstractView {
         }
     }
 
+    public ProgressType view_type { get; construct; }
+
+    public ProgressView (ProgressType type) {
+        Object (view_type: type);
+    }
+
     construct {
-        secondary_label.use_markup = true;
-        secondary_label.label = _("Preparing…");
+        if (view_type == ProgressType.REF_INSTALL) {
+            secondary_label.use_markup = true;
+            secondary_label.label = _("Preparing…");
+        } else if (view_type == ProgressType.BUNDLE_INSTALL) {
+            secondary_label.label = _("Unable to estimate time remaining.");
+        }
```

Comment:
Ah, this doesn't seem to be necessary any more, as both types can share the `Preparing…` placeholder.

```suggestion
        secondary_label.use_markup = true;
        secondary_label.label = _("Preparing…");
```